docs/03-astropy_dev_env.rst

```diff
@@ -0,0 +1,178 @@
+at the bottom of this page.
+
+Step 3: Create a new conda environment for developing astropy
+-------------------------------------------------------------
+
+-  Inside the astropy-dev folder, create a new conda environment called
+   *astropy-dev* and activate it.
+
+   ::
+
+       conda create -n astropy-dev --only-deps python=3 astropy
+       source activate astropy-dev
+       
+   The *--only-deps* flag indicates that the new environment will be installed with all of astropy's dependencies without installing astropy itself.
+
+-  Inside the astropy-dev folder, install the Astropy from the source
+   code:
+
+   ::
+
+       python setup.py develop
+       
+   The *develop* command will install the package in way that does not require you to re-install any time you make changes to your astropy-dev library.
+
+-  At this point, you could install any other packages you normally work
+   with, *except* for astropy. For example,
+
+   ::
+
+       conda install pandas
+
+Step 4: Make a new branch for editing the source code
+-----------------------------------------------------
+
+-  Inside the astropy-dev folder, create a new branch to work in. Try to
+   pick a descriptive name for your branch:
+
+   ::
+
+       git branch my-code-update
+       git checkout my-code-update
+
+Congratulations! You're ready to edit the astropy source code safely.
+Use ``git`` for committing your changes and pushing those changes to
+your ``my-code-update`` branch on Github.
+
+Here's an example of creating a new file and pushing it to the new branch:
+
+::
+
+       echo "print('hello world')" > my-new-code.py
+       git add my-new-code.py
+       git commit -m "first commit"
+       git push origin my-code-update
+
+For more help with learning Git and Github, see the `Github
+Help <https://help.github.com/>`__ pages, the `Git and Github Learning
+Resources <https://help.github.com/articles/git-and-github-learning-resources/>`__
+guide, `try.github.io <http://try.github.io/>`_, and the `Github Learning Lab  <https://lab.github.com/>`_
+
+When you are done working on development, don't forget to exit the
+astropy-dev environment.
+
+::
+
+    source deactivate
+
+When you want your changes to be incorporated into Astropy, `submit a pull request on Github <https://help.github.com/articles/creating-a-pull-request/>`__. If you're looking for `quick ways to contribute to Astropy <http://www.astropy.org/contribute.html#code>`__, check out the issues page on the main Astropy Github repo.
+
+
+Keeping your fork up to date
+============================
+
+If it has been awhile since you last forked the Astropy repo, your fork
+will likely need updating. This tutorial will teach you how to use the
+command line to connect to the main Astropy Github repo, update to the
+latest development version, and push those changes to your personal
+Astropy fork on Github.
+
+Step 1: Add the core Astropy repo to your git config file
```

Comment:
Again this might be overly pedantic... but technically the "git config file" is something different - this is actually adding the core Astropy repo as a "remote" in your local repo (whereas the git config file is shared across all repos on your computer)